### DISCUSSION TITLE
How important is Thirdweb for Web3 Developers? Why does Thirdweb matter?

### DISCUSSION THREAD

**Discussion by heysourin** [Cyfrin/foundry-full-course-cu#61] [General]
I was wondering that developers can write or deploy their codes manually. I know Thirdweb comes with ready-made libraries, codes and SDKs, that save time and also make codes bug-less. 

Rather than that what are the other reasons Thirdweb used for? Can anyone explain?

**Comment by alymurtazamemon** [comment]
@heysourin As you have already mentioned many benefits of it such as easy to use, already built components and written code for the most used smart contracts, a dashboard to deploy, interact and verify contracts.

But one of the important use-case here I found in this course is related to private keys. As you have already known so far that managing private keys is not so efficient so far with the foundry so what we can do is use thirdweb to move our contract to the dashboard and then deploy from there using our metamask. This way we do not need to use the private key.

And yes there are tons of other helpful things thirdweb provides which you should ask the ChatGPT or Bard. Thanks

**Comment by cromewar** [comment]
Also in addition what @alymurtazamemon said, you can always use fully fledged SDK and tools such as third web which will faster your development process, or even other services such as Moralis API to get info out of the blockchain (instead of just using web3  native libraries). But those tools are not perfect, an actual project always need manual verification and auditing. So you can use whatever you like, but the bases are always necessary.

**Comment by ciaranightingale** [comment]
@heysourin  awesome question!
thirdweb is a developer tool that enables you to build web3 projects by helping you to build more efficiently by simplifying the process. This includes prebuilt contracts but also for custom contracts, a CLI (as in this vid) for deploying any contract - which handles unnecessary complexities such as storage, contract verification, and private keys on the contract deployment side. As mentioned above, the Dashboard UI allows you to deploy, interact, and manage your contracts through a frontend (no scrips needed!)
The React and typescript SDKs (and dashboard which is built with the SDK) can recognize what standards are in your contract (e.g. ERC721) and provide smart hooks specific to you contract so you can easily build a front end in only a few lines of code (e.g. useOwnedNfts)
There are lots of other SDKs as well including Solidity (with prebuilt contract add ons/Extensions), Unity(!!) React Native, Python, Go and even a wallet SDK which makes it easy to build ERC-4337 smart contract wallets into your app!
[The docs](https://portal.thirdweb.com/) are also a great source of truth

**Comment by heysourin** [comment]
Thank you @cromewar @ciaranightingale
